### PR TITLE
load_data: fix the bug that column list does not work in load data.

### LINKED ISSUE #35198: Column list does not work in load data statement
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```sql
create table t66 (id int primary key,k int,c varchar(10),dt date,vv char(1),ts datetime);
LOAD DATA LOCAL INFILE '1.csv' INTO TABLE t66 FIELDS TERMINATED BY ',' ENCLOSED BY '\"' IGNORE 1 LINES (k,id,c,dt,vv,ts);
select * from t66;
```
The 1.csv is,
```
dsadasdas
"1","1",,"2022-04-19","a","2022-04-19 00:00:01"
"1","2","a","2022-04-19","a","2022-04-19 00:00:01"
"1","3","a","2022-04-19","a","2022-04-19 00:00:01"
"1","4","a","2022-04-19","a","2022-04-19 00:00:01"
```

### 2. What did you expect to see? (Required)
1. Load data statement should not report warnings;
2. Should get the results
```
mysql> select * from t66;
+----+------+------+------------+------+---------------------+
| id | k    | c    | dt         | vv   | ts                  |
+----+------+------+------------+------+---------------------+
|  1 |    1 |      | 2022-04-19 | a    | 2022-04-19 00:00:01 |
|  2 |    1 | a    | 2022-04-19 | a    | 2022-04-19 00:00:01 |
|  3 |    1 | a    | 2022-04-19 | a    | 2022-04-19 00:00:01 |
|  4 |    1 | a    | 2022-04-19 | a    | 2022-04-19 00:00:01 |
+----+------+------+------------+------+---------------------+
```

### 3. What did you see instead (Required)
1. duplicate warnings
2. only one row loaded
```
mysql> select * from t66;
+----+------+------+------------+------+---------------------+
| id | k    | c    | dt         | vv   | ts                  |
+----+------+------+------------+------+---------------------+
|  1 |    1 |      | 2022-04-19 | a    | 2022-04-19 00:00:01 |
+----+------+------+------------+------+---------------------+
1 row in set (0.00 sec)
```

### 4. What is your TiDB version? (Required)
v5.0.0 and above should be affected, because it is introduced by https://github.com/pingcap/tidb/pull/18753 .

### PR DESCRIPTION
**Pull request by SpadeA-Tang**
### What problem does this PR solve?

Issue Number: close #35198

Problem Summary:

### What is changed and how it works?

As the issue describes, the column list does not work in load data. 

This is caused by that in `initLoadColumns`, `e.insertColumns` is appended according to the order of the table columns, not the order of the column list.

### Check List

Tests 

- [x] Integration test

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Fix the issue that column list does not work in load data.
```

### CODE REVIEW

**Review by jackysp — COMMENTED** [review]
I think this bug-fix needs a release note.

**Comment by SpadeA-Tang** [comment]
> I think this bug-fix needs a release note.

I have added the release note.

File: executor/load_data.go [hunk 1]
```diff
@@ -129,6 +129,36 @@ type FieldMapping struct {
...
+		return nil
+	}
+
+	mapping := make(map[string]int)
+	for idx, colName := range columnNames {
+		mapping[colName] = idx
+	}
+
+	for _, col := range cols {
+		idx := mapping[col.Name.O]
```
Comment (crazycs520) [inline executor/load_data.go:153]: Should use `col.Name.L`? Also in line#149 should use `strings.ToLower(colName)` as a key?

Please add a test for this case.

Comment (SpadeA-Tang) [reply executor/load_data.go:153]: Good catch, we should convert them to lower case. I will modify it immediately.

File: executor/load_data.go [hunk 2]
```diff
@@ -129,6 +129,36 @@ type FieldMapping struct {
 	UserVar *ast.VariableExpr
 }
 
+// reorderColumns reorder the e.insertColumns according to the order of columnNames
+// Note: We must ensure there must be one-to-one mapping between e.insertColumns and columnNames in terms of column name.
+func (e *LoadDataInfo) reorderColumns(columnNames []string) error {
+	cols := e.insertColumns
+
+	if len(cols) != len(columnNames) {
+		return errors.New("LOAD DATA: columns unmatched")
```
[the remark this answers is not in the document]
Comment (SpadeA-Tang) [reply executor/load_data.go]: Done.

**Comment by SpadeA-Tang** [comment]
PTAL @jackysp @crazycs520

**Review by jackysp — APPROVED** [review]
LGTM

**Review by youjiali1995 — APPROVED** [review]

**Pull request merged**